test/e2e/gameserver_test.go

```diff
@@ -442,28 +446,44 @@ func TestDevelopmentGameServerLifecycle(t *testing.T) {
 			},
 		},
 	}
-	readyGs, err := framework.CreateGameServerAndWaitUntilReady(t, framework.Namespace, gs)
+	readyGs, err := framework.CreateGameServerAndWaitUntilReady(t, framework.Namespace, gs.DeepCopy())
 	if err != nil {
 		t.Fatalf("Could not get a GameServer ready: %v", err)
 	}
-
-	assert.Equal(t, readyGs.Status.State, agonesv1.GameServerStateReady)
+	require.Equal(t, readyGs.Status.State, agonesv1.GameServerStateReady)
 
 	// confirm delete works, because if the finalisers don't get removed, this won't work.
 	err = framework.AgonesClient.AgonesV1().GameServers(framework.Namespace).Delete(ctx, readyGs.ObjectMeta.Name, metav1.DeleteOptions{})
-	assert.NoError(t, err)
+	require.NoError(t, err)
 
 	err = wait.PollImmediate(time.Second, 10*time.Second, func() (bool, error) {
 		_, err = framework.AgonesClient.AgonesV1().GameServers(framework.Namespace).Get(ctx, readyGs.ObjectMeta.Name, metav1.GetOptions{})
-
 		if k8serrors.IsNotFound(err) {
 			return true, nil
 		}
 
 		return false, err
 	})
+	require.NoError(t, err)
 
-	assert.NoError(t, err)
+	// let's make sure we can allocate a dev gameserver
```

Comment:
I'm curious why you chose to do the allocation test on a second game server instead of allocating the one above prior to deleting it. It seems like allocation the one above would make the test run faster and provide the same coverage.

Reply:
Since there was a question of making sure delete still works, I wanted to check both Ready and Allocated would delete as expected just to be extra careful.

That being said, I realise I didn't check for deletion of the allocation state, so I'm adding that now.